RSSTimeFormatter/RealDateTimeFormatter.cs

```diff
@@ -208,17 +214,29 @@ public string PrintDateDeltaCompact(double time, bool includeTime, bool includeS
             return sb.ToStringAndRelease();
         }
 
+        private string FormatDate(DateTime t)
+        {
+            return string.Format(culture, "{0:" + dateFormat + "}", t);

```

Comment:
> "{0:" + dateFormat + "}"

Those formats should get cached to save 3 allocations. Date formatting will get hammered quite a lot from KSP itself and various mods.